tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -45,7 +45,11 @@ namespace tensorflow {
 namespace grappler {
 
 #if TENSORFLOW_USE_ROCM
+#if TF_ROCM_VERSION < 30900
 const std::array<std::string, 2> FP16SupportedDevices = {"906", "908"}; 
+#else
+const std::array<std::string, 2> FP16SupportedDevices = {"gfx906", "gfx908"}; 
```

Comment:
This is not going to have any effect. The code will try to find the string "906" in this array (see lines 55-58) and will fail.

Besides, it does not even address the fundamental problem: HasEnhancedFP16ComputeSupport() does not receive {906,0}, it receives {0,0}, because GetDeviceGPUArch() (line 1153) expects the "architecture" string to be a sequence of numbers separated by dots, and it is not.

I traced that string back to the origin and it is set here https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/core/grappler/clusters/utils.cc#L144 - that code is over a year old, so the issue is not related to 3.9 or any other version of ROCm. The AMP code is simply broken. 

The bit that determines architecture version for ROCm never worked correctly in the first place, but, in the past, it did not matter, because kMinGPUArch was {0, 0} and AutoMixedPrecisionImpl::IsOnSuitableGPUArch() always returned true (see https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/de688d709e852c31274092407d469a880b8c66f5/tensorflow/core/grappler/optimizers/auto_mixed_precision.cc).

Then, this commit https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/9544f1095f73050d44ed4757411ea6a0d92dde28 changed kMinGPUArch to {906, 0}, and ever since then AMP was effectively disabled, because AutoMixedPrecisionImpl::IsOnSuitableGPUArch() returns false for all ops and on all architectures (and also because GetNumGPUs(), which depends on the same GetDeviceGPUArch(), always returns 0.)

Reply:
@ekuznetsov139 Thanks for looking into this. I'll see how I can update grappler/clusters/utils to give us the expected result.